OpenKh.Tools.Kh2TextEditor/ViewModels/TextEditorViewModel.cs

```diff
@@ -127,6 +139,66 @@ public TextEditorViewModel()
             Drawing = new SpriteDrawingDirect3D();
             CurrentMessageEncoder = Encoders.InternationalSystem;
             _messages = new MessagesModel(this, this, new Msg.Entry[] { });
+
+            HandleReturn = new RelayCommand(x =>
+            {
+                Text += "{:newline}";                
+                SelectedItem = SelectedItem;
+            });
+
+            HandleRemoval = new RelayCommand(x =>
+            {
+                MessageEntries.RemoveAll(x => x.Id == SelectedItem.Id);
+                ResetMessagesView();
+            });
+
+            HandleAddition = new RelayCommand(x =>
+            {
+                int _id = MessageEntries[MessageEntries.Count - 1].Id;
```

Comment:
This assumes that the list is always ordered at that there is at least one element. I would suggest to do the more secure `MessageEntries.Max(x => x.Id) + 1`.

I do not know how much important this is, but IDs between the various MSG files never overlaps, even if they belong to different worlds. I have no idea why the developers took such decision as multiple MSGs are never load. I would not be bothered by this now, but that can be considered in the future.

Reply:
Roger. Would be safer to do this as mentioned.